build_tools/teatime.py

```diff
@@ -103,6 +144,79 @@ def finish(self):
                 b"[" + self.label + b" completed in " + run_pretty.encode() + b"]\n"
             )
 
+        # Call coordinate_index_and_logs if indexing is enabled.
+        if (
+            not self.skip_index
+            and self.upload_to_s3
+            and self.s3_bucket
+            and self.log_dir
+        ):
+            self.coordinate_index_and_logs()
+
+    def coordinate_index_and_logs(self):
+        # Determine paths
+        log_dir = self.log_dir
+        amdgpu_family = os.getenv("AMDGPU_FAMILIES")
+
+        # Step 1: Run create_log_index.py (if AMDGPU_FAMILIES is defined)
+        if amdgpu_family:
+            try:
+                index_script = (
+                    Path(__file__).resolve().parent.parent
+                    / "build_tools"
+                    / "create_log_index.py"
+                )
+                print(f"[TEATIME] Indexing logs for AMDGPU_FAMILIES={amdgpu_family}")
+                subprocess.run(
+                    [
+                        sys.executable,
+                        str(index_script),
+                        "--log-dir",
+                        str(log_dir),
+                        "--amdgpu-family",
+                        amdgpu_family,
+                    ],
+                    check=True,
+                )
+            except subprocess.CalledProcessError as e:
+                print(f"[WARN] create_log_index.py failed: {e}", file=sys.stderr)
+            except Exception as e:
+                print(
+                    f"[WARN] Unexpected error during log indexing: {e}", file=sys.stderr
+                )
+        else:
+            print("[WARN] AMDGPU_FAMILIES not set; skipping log indexing")
+
+        # Step 2: S3 upload using --bucket and --subdir
+        try:
+            upload_script = (
+                Path(__file__).resolve().parent.parent
+                / "build_tools"
+                / "upload_logs_to_s3.py"
+            )
+            print(f"[TEATIME] Uploading logs to s3://{self.s3_bucket}/{self.s3_subdir}")
+            subprocess.run(
+                [
+                    sys.executable,
+                    str(upload_script),
+                    "--log-dir",
+                    str(log_dir),
+                    "--bucket",
+                    self.s3_bucket,
+                    "--subdir",
+                    self.s3_subdir,
+                ],
+                check=True,
+            )
+        except subprocess.CalledProcessError as e:
+            print(f"[WARN] Log upload failed: {e}", file=sys.stderr)
+            if os.getenv("TEATIME_FAIL_ON_UPLOAD_ERROR") == "1":
+                raise
+        except Exception as e:
+            print(f"[WARN] Unexpected error during log upload: {e}", file=sys.stderr)
+            if os.getenv("TEATIME_FAIL_ON_UPLOAD_ERROR") == "1":
+                raise
```

Comment:
We don't need to `subprocess.run` Python scripts we control from other Python scripts we control. Could switch to a Python library model where we import the file and call a function directly.